hera_cal/utils.py

```diff
@@ -1244,14 +1260,24 @@ def red_average(data, reds=None, bl_tol=1.0, inplace=False,
                 n = np.asarray([data.get_nsamples(bl + (pol,)) for bl in blg])
                 tint = np.asarray([data.integration_time[data.antpair2ind(bl + (pol,))] for bl in blg])[:, :, None]
                 w = np.asarray([wgts[bl + (pol,)] for bl in blg]) * tint
-
+            if propagate_flags:
+                # if propagate flags is True, then set all of w to zero on fully flagged baselines
+                # even if the weights were provided by the user.
+                w[np.asarray([np.all(np.isclose(fslice, 0.0)) for fslice in f])][:] = 0.0
```

Comment:
> I think this is wrong. The point of the `wgts` dictionary is to have _total control_ of `w` by the user. we should not be manipulating `wgts` if they are fed. `propagate_flags` is simply going to change the `favg` output, but should not change `wgts`.
> 
> I see why you did this though, which is because in practice, you (@anguta), want the specific functionality of not setting `w->0` when all baselines are flagged. Therefore, setting `wgts = n * ~f` and then feeding `red_average(..., wgts=wgts)` would not accomplish this, and needs to be "unflagged" where `f` is all True in order for the wgts to not be zero. However, I think my point here is that _you need to do that yourself_ outside of the function. In my mind, the point of `red_average` (or any general purpose function for that matter) is to support broad functionality, and when you want to implement a specific use-case you should conform your data to make that work outside the function, which in this case is fairly easy anyways, as its a one-liner!

@nkern, this is already implemented outside of the function (setting the weights to zero on fully flagged antennas) so I agree that the switch is not strictly necessary. 


> 
> Another way to say this, is that you are making an assumption about the form of `wgts` that is not necessarily true, i.e., that `wgts = n * ~f`
> 
> I think an alternative if you must have this fully automated is to make an experimental wrapper `_aew_red_average` function. Lmk what you think.